jooby-filewatcher/src/main/java/org/jooby/filewatcher/FileMonitor.java

```diff
@@ -0,0 +1,155 @@
+package org.jooby.filewatcher;
+
+import static java.nio.file.StandardWatchEventKinds.ENTRY_CREATE;
+import static java.nio.file.StandardWatchEventKinds.OVERFLOW;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchEvent.Kind;
+import java.nio.file.WatchEvent.Modifier;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Injector;
+
+class FileMonitor implements Runnable {
+
+  /** The logging system. */
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private Injector injector;
+
+  private WatchService watcher;
+
+  private final Map<WatchKey, Path> keys = new ConcurrentHashMap<>();
+
+  private final Map<Path, FileEventOptions> options = new ConcurrentHashMap<>();
+
+  private Set<FileEventOptions> optionList;
+
+  @Inject
+  public FileMonitor(final Injector injector,
```

Comment:
injector, watcher and optionList fields can be final, is there a reason why are they not final while some other non-static fields final?